core/objects.go

```diff
@@ -219,6 +219,8 @@ type Challenge struct {
 	// Contains information about URLs used or redirected to and IPs resolved and
 	// used
 	ValidationRecord []ValidationRecord `json:"validationRecord,omitempty"`
+	// RFC3339 timestamp. Required by RFC8555 if status is valid
+	Validated string `json:"validated,omitempty"`
```

Comment:
I would prefer to take the approach taken by the Authorization object below, where an omit-able timestamp is represented by `*time.Time`, rather than by `string`:

https://github.com/letsencrypt/boulder/blob/main/core/objects.go#L347-L352

Although the ideal of updating this type from a string to a time later is a good one, changing the type of things that are part of our gRPC API (let alone part of our public-facing API) is very hard, so it's best to get this right on the first try.

Reply:
Thanks, I like that better too! I have pushed a new change up to use *time.Time